### ISSUE TITLE
endless Waiting for shard 0 to be ready... when using only 1 GPU (google/flan-t5-xl)

### ISSUE THREAD

**Issue by captainst** [huggingface/text-generation-inference#471]
### System Info

ubuntu 20.04, CUDA 11.7, pytorch 2.0.1
text-generation-launcher docker latest

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

1. Download the huggingface google/flan-t5-xl model to your local drive (for example, _$PWD/models/flan-t5-xl_)
2.  export CUDA_VISIBLE_DEVICES=0
3. docker run --gpus 0 -p 8080:80 -v $PWD/models/flan-t5-xl:/data ghcr.io/huggingface/text-generation-inference:latest  --model-id google/flan-t5-xl --sharded false  --weights-cache-override /data

I see  Waiting for shard 0 to be ready... for ever. Both GPUs are idle.

**However, if I change to use 2 gpus, with --num-shard 2, the model is loaded and distributed to 2 GPUs.**

### Expected behavior

I suppose that the docker image should work also with 1 GPU without sharding.

**Comment by captainst** [comment]
@OlivierDehaene Thank you! If I understood correctly the modifications, the issue was that **Seq2SeqLM** does NOT support loading weights from local harddrive, right ?

**Comment by OlivierDehaene** [comment]
Seq2SeqLM loads on CPU first before sending the weights to the CUDA device. It takes ages to load to CPU.
With the fix, T5 skips this by loading directly to the correct device using safetensors. The problem still exists for other seq2seqlms.

**Comment by ashwin-js** [comment]
Hi I am still getting this error when I try with Flan-ul2 in sagemaker.huggingface.

`import json
import sagemaker
import boto3
from sagemaker.huggingface import HuggingFaceModel, get_huggingface_llm_image_uri

try:
	role = sagemaker.get_execution_role()
except ValueError:
	iam = boto3.client('iam')
	role = iam.get_role(RoleName='sagemaker_execution_role')['Role']['Arn']

# Hub Model configuration. https://huggingface.co/models
hub = {
	'HF_MODEL_ID':'google/flan-ul2',
	'SM_NUM_GPUS': json.dumps(1)
}

# create Hugging Face Model Class
huggingface_model = HuggingFaceModel(
	image_uri=get_huggingface_llm_image_uri("huggingface",version="0.8.2"),
	env=hub,
	role=role, 
)

# deploy model to SageMaker Inference
predictor = huggingface_model.deploy(
	initial_instance_count=1,
	instance_type="ml.g5.2xlarge",
	container_startup_health_check_timeout=300,
  )
  
# send request
predictor.predict({
	"inputs": "Translate to German:  My name is Arthur",
})`

**Comment by Narsil** [comment]
Sagemaker is still on 0.8.2 I think.

0.9.3 should be coming soon with the fix hopefully @philschmid Is that correct ?

**Comment by philschmid** [comment]
Yes working on the release of 0.9.3. I hope i can report something good EOW.
